aydin/nn/models/torch/torch_jinet.py

```diff
@@ -0,0 +1,274 @@
+                    1,
+                    1,
+                    self.spacetime_ndim,
+                    padding=dilation * radius,
+                    kernel_size=kernel_size,
+                    dilation=dilation,
+                    activation="lrel",
+                )
+            )
+
+            # update receptive field radius
+            current_receptive_field_radius += dilation * radius
+
+        if spacetime_ndim == 2:
+            self.conv = nn.Conv2d
+        elif spacetime_ndim == 3:
+            self.conv = nn.Conv3d
+        else:
+            raise ValueError("spacetime_ndim can not be anything other than 2 or 3...")
+
+        if self.nb_channels is None:
+            self.nb_channels = sum(self.num_features) * 2
+
+        nb_out = 1
+        self.kernel_one_conv_functions = []
+        for index in range(len(self.kernel_sizes)):
+            nb_in = nb_out
+            nb_out = (
+                self.nb_out_channels
+                if index == (self.nb_dense_layers - 1)
+                else self.nb_channels
+            )
+
+            self.kernel_one_conv_functions.append(
+                self.conv(
+                    nb_in,
+                    nb_out,
+                    kernel_size=1,
+                    padding="valid",
+                )
+            )
+
+        self.final_kernel_one_conv = self.conv(
+            nb_out,
+            self.nb_output_ch,
+            kernel_size=(1,) * spacetime_ndim,
+            padding_mode="same",
+        )
+
+        self.relu = nn.ReLU()
+        self.lrelu = nn.LeakyReLU(negative_slope=0.01)
+
+    @property
+    def kernel_sizes(self):
+        if self._kernel_sizes is None:
+            if self.spacetime_ndim == 2:
+                self._kernel_sizes = [7, 5, 3, 3, 3, 3, 3, 3]
+            elif self.spacetime_ndim == 3:
+                self._kernel_sizes = [7, 5, 3, 3]
+
+        return self._kernel_sizes
+
+    @property
+    def num_features(self):
+        if self._num_features is None:
+            if self.spacetime_ndim == 2:
+                self._num_features = [64, 32, 16, 8, 4, 2, 1, 1]
+            elif self.spacetime_ndim == 3:
+                self._num_features = [10, 8, 4, 2]
+
+        return self._num_features
+
+    def forward(self, x):
+        dilated_conv_list = []
+
+        # Calculate dilated convolutions
+        for index in range(len(self.kernel_sizes)):
+            x = self.dilated_conv_functions[index](x)
+            dilated_conv_list.append(x)
+
+        # Concat the results
+        x = torch.cat(dilated_conv_list, dim=-1)  # TODO: pass axis as -1
+
+        # First kernel size one conv
+        x = self.kernel_one_conv_functions[0](x)
+        x = self.lrelu(x)
+        y = x
+
+        # Rest of the kernel size one convolutions
+        for index in range(1, self.nb_dense_layers + 1):
+            x = self.kernel_one_conv_functions[index](x)
+            x = self.lrelu(x)
+            y += self.f * x
```

Comment:
I think we can find a better variable name to `self.f`. @li-li-github how else can we name this variable? This is basically equal to this: https://github.com/royerlab/aydin/blob/af0756b09863c2617283ed9ac917b5f6734120e6/aydin/nn/models/jinet.py#L249

Reply:
removed